pkgs/by-name/cr/creality-print/package.nix

```diff
@@ -0,0 +1,69 @@
+{
+  lib,
+  stdenv,
+  undmg,
+  fetchurl,
+  makeDesktopItem,
+  appimageTools,
+}:
+let
+  pname = "creality-print";
+  version = if stdenv.isLinux then "4.3.7.6627" else "5.1.1.9490";
+  versionChangelog = lib.concatStringsSep "." (lib.take 3 (lib.splitString "." version));
+
+  src = fetchurl {
```

Comment:
Rather than the confusing-to-read and less-maintainable chain of if-statements, why not move the invidual `src` values into the respective linux and darwin derivations? I imagine that'd be friendlier to existing update scripts as well.